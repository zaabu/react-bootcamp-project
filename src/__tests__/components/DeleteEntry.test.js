import React from "react";
import { shallow, configure } from "enzyme";
import configureMockStore from "redux-mock-store";
import  DeleteEntryConfirm  from "../../components/entries/deleteDiaryEntry";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });


describe("SignupUser component", () => {
  const props = {
    handleDelete: jest.fn(),
  };
  const mockStore = configureMockStore();
  let wrapper;

  beforeEach(() => {
    const store = mockStore({});
    wrapper = shallow(<DeleteEntryConfirm {...props} store={store} />);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
