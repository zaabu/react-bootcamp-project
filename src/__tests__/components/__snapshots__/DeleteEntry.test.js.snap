// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupUser component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteEntryConfirm
    handleDelete={[MockFunction]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Delete",
      "class": "btn btn-danger",
      "href": "#",
      "id": "danger-button",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": "Delete",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Delete",
        "class": "btn btn-danger",
        "href": "#",
        "id": "danger-button",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "Delete",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
